assets/js/modules/analytics/common/use-snippet-switch.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Analytics Use Snippet Switch component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../datastore/constants';
+import Switch from '../../../components/switch';
+const { useSelect, useDispatch } = Data;
+
+export default function UseSnippetSwitch() {
+	const useSnippet = useSelect( ( select ) => select( STORE_NAME ).getUseSnippet() );
+
+	const { setUseSnippet } = useDispatch( STORE_NAME );
+	const onChange = useCallback( () => {
+		setUseSnippet( ! useSnippet );
+	}, [ useSnippet ] );
+
+	return (
+		<div className="googlesitekit-analytics-usesnippet">
+			<Switch
+				label={ __( 'Let Site Kit place code on your site', 'google-site-kit' ) }
+				checked={ useSnippet }
```

Comment:
A loading state or at least setting `disabled={ useSnippet === undefined }` here would be good; otherwise a user could change it while loading, then it might change back once the data from the server is loaded… they wouldn't be getting an accurate view of their data from this view if we treat `undefined === false`.

At least disabling this control until we have the data from settings would be good.

Reply:
I'd say we should return nothing if it is `undefined`. We handle this at a higher level component, so it doesn't matter at all for the user, but we should definitely build safeguards for "non-loaded" data into all these components. I'd say it's okay just returning nothing for the cases where there's no proper loading indicator and the cases are handled by the parent component anyway.

Reply:
✅